internal/tweets/webapi/twitter.go

```diff
@@ -2,31 +2,83 @@ package webapi
 
 import (
 	"context"
-
+	"encoding/json"
+	"fmt"
 	"github.com/kordape/tweety/internal/entity"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// testing path
+// http://localhost:8080/v1/tweets/classify?userId=1277254376&maxResults=90&startTime=2022-01-12&endTime=2022-06-15
+
+const (
+	getUsersTweetsUrl = "https://api.twitter.com/2/users/%s/tweets/"
 )
 
 type TwitterWebAPI struct {
-	accessKey string
-	secretKey string
+	bearerToken string
 }
 
-func New(accessKey string, secretKey string) *TwitterWebAPI {
+func New(bearerToken string) *TwitterWebAPI {
 	return &TwitterWebAPI{
-		accessKey: accessKey,
-		secretKey: secretKey,
+		bearerToken: bearerToken,
 	}
 }
 
-func (t *TwitterWebAPI) FetchTweets(ctx context.Context, pageId string) ([]entity.Tweet, error) {
-	// TODO: call twitter api to fetch latest tweets from page with pageId
-	// parse tweets into internal struct
+func (t *TwitterWebAPI) FetchTweets(ctx context.Context, userId string, maxResults int, startTime, endTime *time.Time) ([]entity.Tweet, error) {
+	//TODO: add corresponding error messages
+	// refactor and optimise code additionally
 
-	tweet1 := entity.Tweet{
-		Text: "dummy tweet1",
+	baseUrl := fmt.Sprintf(getUsersTweetsUrl, userId)
+	var queryParams []string
+	// max_results query param for setting the number of tweets to be returned: min=5 max =100
+	queryParams = append(queryParams, fmt.Sprintf("max_results=%d", maxResults))
+	queryParams = append(queryParams, "tweet.fields=id,text,created_at")
+	if startTime != nil {
+		tmp := *startTime
+		queryParams = append(queryParams, fmt.Sprintf("start_time=%s", tmp.Format(time.RFC3339)))
 	}
-	tweet2 := entity.Tweet{
-		Text: "dummy twee2",
+	if endTime != nil {
+		tmp := *endTime
+		queryParams = append(queryParams, fmt.Sprintf("end_time=%s", tmp.Format(time.RFC3339)))
 	}
-	return []entity.Tweet{tweet1, tweet2}, nil
+	url := fmt.Sprintf("%s?%s", baseUrl, strings.Join(queryParams, "&"))
+	httpClient := http.Client{}
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+	request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", t.bearerToken))
+	resp, err := httpClient.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("error doing request: %w", err)
+	}
+
+	defer resp.Body.Close()
+	response, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading response: %w", err)
+	}
+
+	var tweeterResponse getUserTweetsResponse
+	err = json.Unmarshal(response, &tweeterResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	return tweeterResponse.Data, nil
+}
+
+type getUserTweetsResponse struct {
+	Data []entity.Tweet `json:"data"`
```

Comment:
mesas domene `entity` i `webapi` time sto si embeddovao ovaj `Data`. Bolje je da imas odvojenu strukturu sa response i da onda radis konverziju u `entity.Tweet` sto vracas kao response.

Evo ti primer sto: zamisli da zelimo da izbacimo `create_at` iz `entity.Tweet` ali hocemo da ga procitamo iz response-a i uradimo neku validaciju (npr). Ne bi mogao to da uradis sa ovim pristupom.

Reply:
`Data` je ime polja tipa `entity.Tweet`. Kad kazem embeddovan mislim da imas dependecy na taj tip

Reply:
aha jasno :)

Reply:
sredjeno